src/dataprocessor/xmldataprocessor.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import BasicHtmlWriter from './basichtmlwriter.js';
+import DomConverter from '../view/domconverter.js';
+import { NBSP_FILLER } from '../view/filler.js';
+
+/**
+ * XmlDataProcessor class.
+ * This data processor implementation uses XML as input/output data.
```

Comment:
Explanation what is the difference between  XML and HTML data processor would be nice, maybe with some examples. I believe this is a question everybody will ask, checking this class: why we need both HTML and XML DP.